osgp/shared/osgp-throttling-client/src/main/java/org/opensmartgridplatform/throttling/ThrottlingClient.java

```diff
@@ -248,6 +280,11 @@ public Permit requestPermitUsingNetworkSegmentIfIdsAreAvailable(
   }
 
   private Integer numberOfGrantedPermits(final int requestId) {
+    if (!this.register()) {
+      LOGGER.error(
+          "Client could not be registered before requesting permit using requestId {}", requestId);
```

Comment:
In line with the other comment about yet, it might be better to change "Client could not be registered before" into something like "Client is not registered when" in the messages if `register` returns `false`.
It may not be likely with the current way the `ThrottlingClient` is used, but if some later change would trigger this scenario the remark that registration has not happened could become misleading.

Reply:
Changed 3 times

Reply:
Why only with the `numberOfGrantedPermits` and one of the `releasePermit` methods, and not with the other `releasePermit` and `discardPermit`?

Reply:
Changed